app/background/events/cosmeticFilterEvents.ts

```diff
@@ -0,0 +1,87 @@
+// import windowActions from '../actions/cosmeticFilterActions'
+// import tabActions from '../actions/tabActions'
+
+// import * as cosmeticFilterAPI from '../api/cosmeticFilterAPI'
+// background
+
+// this definitely goes into state
+import cosmeticFilterActions from '../actions/cosmeticFilterActions'
+
+let rule = {
+  host: '',
+  selector: ''
+}
+
+// add context menu
+chrome.runtime.onInstalled.addListener(function () {
+
+  // parent menu
+  chrome.contextMenus.create({
+    title: 'Brave',
+    id: 'brave',
+    contexts: ['all']
+  })
+
+  // block ad child menu
+  chrome.contextMenus.create({
+    title: 'Block ad',
+    id: 'addBlockElement',
+    parentId: 'brave',
+    contexts: ['all']
+  }, () => {
+    console.log('block custom ad sub-context menu created')
+  })
+
+  chrome.contextMenus.create({
+    title: 'Reset site filter settings',
+    id: 'resetSiteFilterSettings',
+    parentId: 'brave',
+    contexts: ['all']
+  }, () => {
+    console.log('reset site cosmetic filter sub-context menu created')
+  })
+
+  chrome.contextMenus.create({
+    title: 'Reset all filter settings',
+    id: 'resetAllFilterSettings',
+    parentId: 'brave',
+    contexts: ['all']
+  }, () => {
+    console.log('log chrome extension storage menu created')
+  })
+})
+
+// contextMenu listener - when triggered, grab latest selector
+chrome.contextMenus.onClicked.addListener(function (info, tab) {
+  switch (info.menuItemId) {
+    case 'addBlockElement':
+      {
+        rule.selector = window.prompt('rule: ', `${rule.selector}`) || ''
+        chrome.tabs.insertCSS({
+          code: `${rule.selector} {display: none;}`
+        })
+        cosmeticFilterActions.siteCosmeticFilterAdded(rule.host, rule.selector)
+        break
+      }
+    case 'resetSiteFilterSettings':
+      {
+        cosmeticFilterActions.siteCosmeticFilterRemoved(rule.host)
+        break
+      }
+    case 'resetAllFilterSettings':
+      {
+        cosmeticFilterActions.allCosmeticFiltersRemoved()
+        break
+      }
+    default: {
+      console.log('invalid context menu option - cosmeticFilterEvents.ts')
+    }
+  }
+})
+
+// content script listener for right click DOM selection event
+chrome.runtime.onMessage.addListener((msg, sender, sendResponse) => {
+  rule.host = msg.baseURI
```

Comment:
should the rule host include port number?

Reply:
I propose filing that as a separate issue and keeping it as-is for now as there are several other features that have higher priority (mutation observer, UI > port for now).